lib/runtime/goto.js

```diff
@@ -2,20 +2,55 @@
 
 import path from 'path'
 import fs from 'fs'
+import { CompositeDisposable } from 'atom'
 
 import { client } from '../connection'
 import modules from './modules'
 import { isValidScopeToInspect } from '../misc/scopes'
 import { getWord, getWordRangeAtBufferPosition, isValidWordToInspect } from '../misc/words'
+import { getLocalContext } from '../misc/blocks'
 
-const methods = client.import('methods')
+const goto = client.import('goto')
 
+const grammar = atom.grammars.grammarForScopeName('source.julia')
 const includeRegex = /(include|include_dependency)\(".+\.jl"\)/
 const filePathRegex = /".+\.jl"/
 
 class Goto {
   activate (ink) {
     this.ink = ink
+    this.subscriptions = new CompositeDisposable()
+    this.refreshFiles = new Set()
+    this.subscriptions.add(
+      atom.workspace.observeTextEditors(editor => {
+        if (
+          !this.refreshFiles.has(editor.getPath()) &&
+          editor.getGrammar() === grammar
+        ) {
+          const buffer = editor.getBuffer()
+          const bufferSubscriptions = new CompositeDisposable()
+          bufferSubscriptions.add(buffer.onDidSave(({ path }) => {
+            if (!this.refreshFiles.has(path)) {
+              this.refreshFiles.add(path)
```

Comment:
You can just unconditionally `add(path)` here, no? `refreshFiles` is a `Set` after all, so adding twice doesn't hurt.

Reply:
year, maybe I forget what `Set` is for at all xD